src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -52,78 +61,103 @@ public DrsService(SnapshotService snapshotService,
                       DrsIdService drsIdService,
                       IamService samService,
                       ResourceService resourceService,
+                      ConfigurationService configurationService,
+                      KubeService kubeService,
                       PerformanceLogger performanceLogger) {
         this.snapshotService = snapshotService;
         this.fileService = fileService;
         this.drsIdService = drsIdService;
         this.samService = samService;
         this.resourceService = resourceService;
+        this.configurationService = configurationService;
+        this.kubeService = kubeService;
         this.performanceLogger = performanceLogger;
     }
 
-    public DRSObject lookupObjectByDrsId(AuthenticatedUserRequest authUser, String drsObjectId, Boolean expand) {
-
-        DrsId drsId = drsIdService.fromObjectId(drsObjectId);
-        SnapshotProject snapshotProject = null;
-        try {
-            UUID snapshotId = UUID.fromString(drsId.getSnapshotId());
-            // We only look up DRS ids for unlocked snapshots.
-            String retrieveTimer = performanceLogger.timerStart();
-
-            snapshotProject = snapshotService.retrieveAvailableSnapshotProject(snapshotId);
-
-            performanceLogger.timerEndAndLog(
-                retrieveTimer,
-                drsObjectId, // not a flight, so no job id
-                this.getClass().getName(),
-                "snapshotService.retrieveAvailable");
-        } catch (IllegalArgumentException ex) {
-            throw new InvalidDrsIdException("Invalid object id format '" + drsObjectId + "'", ex);
-        } catch (SnapshotNotFoundException ex) {
-            throw new DrsObjectNotFoundException("No snapshot found for DRS object id '" + drsObjectId + "'", ex);
+    private class DrsRequestResource implements AutoCloseable {
+        DrsRequestResource() {
+            // make sure not too many requests are being made at once
+            int podCount = kubeService.getActivePodCount();
+            int maxDRSLookups = configurationService.getParameterValue(ConfigEnum.DRS_LOOKUP_MAX);
+            int max = maxDRSLookups / podCount;
+            if (currentDRSRequests.incrementAndGet() > max) {
```

Comment:
needs to be `get` otherwise this will increment the counter twice
```suggestion
            if (currentDRSRequests.get() > max) {
```